### PR TITLE
Add Passport widget in nav bar in Explorer

### LINKED ISSUE #2433: Explorer: add Passport widget in nav bar
**User Story:**
As a donor 
I want to know what my Passport score and donation impact are 
So that I know the impact of my donation 

**Acceptance Criteria**

_**Empty State**_
GIVEN that I don't have a Passport or have a score of zero 
WHEN I look at the nav bar on Explorer 
THEN I see a Passport widget with "No score detected" copy 

GIVEN that I don't have a Passport or have a score of zero
WHEN I click on the widget 
THEN I see copy about no score detected 
AND a CTA to go to Passport

**_Happy Path_**
GIVEN that I have a nonzero Passport score 
WHEN I look at the nav bar on Explorer 
THEN I see a Passport widget with my score 

GIVEN that I have a nonzero Passport score 
WHEN I click on the widget 
THEN I see more information about my Passport score 
AND my donation impact percentage
AND a CTA to go to Passport

**Product & Design Links:**
* [Figma](here are the updates to the passport widget (https://www.figma.com/file/1uxgU2jb0yyN11SUgxARmF/Explorer-S19?type=design&node-id=719-30288&mode=design&t=HhcD8Go9ILxSVVUZ-4)

* The color of your score and donation impact percentage depends on your score 

Matching framework: 
0-15: 0% Match
15-25: 0-100% Match

Color system: 
> Score of 0-15: Red 
> Score of 15-25: Yellow 
> Score of 25+: Green 

**Tech Details:**

**Open Questions:**

**Notes/Assumptions:**
- This includes writing test coverage for the functionality added in the ticket, if necessary

### LINKED ISSUE #2460: Explorer: Passport cleanup to be consistent with new Passport sliding scale feature
**Context**: Let's remove the old Passport flow in Explorer to be consistent with the new Passport sliding scale feature. 

**Acceptance Criteria**
- [ ] Remove Passport banner flow 
- [ ] Remove /passport page on Explorer 
- [ ] Remove the "Go to Passport" modal when you're checking out projects (see this ticket for more context: https://github.com/gitcoinco/grants-stack/issues/2268) 

**Product & Design Links:**

**Tech Details:**

**Open Questions:**

**Notes/Assumptions:**
- This includes writing test coverage for the functionality added in the ticket, if necessary

### PR DESCRIPTION
**Pull request by bhargavaparoksham**
## PR checklist

For every PR, make sure that these statements are true:

- [x] Includes only changes relevant to the original ticket. Significant refactoring needs to be separated.
- [x] Doesn't contain type casts and non-null assertions.
- [x] Doesn't add `@ts-ignore`.
- [x] Doesn't disable lints.
- [x] Doesn't use `useState` just for computation - use plain variables instead.
- [x] Splits components into pure components that don't depend on external state or hooks.
- [x] Avoid embedding components within other components
- [x] Doesn't propagate optional values without good reason, doesn't mark property values as optional if that doesn't represent reality.
- [x] Doesn't duplicate existing code.
- [x] Parses out-of-domain data - this includes user input, API respones, on-chain data etc.
- [x] Doesn't contain commented out code.
- [x] Doesn't contain skipped or empty tests.
- [x] If this PR adds/updates any feature, it adds/updates its test script

##### Description

As a donor
I want to know what my Passport score and donation impact are
So that I know the impact of my donation

Remove the old Passport flow in Explorer to be consistent with the new Passport sliding scale feature.

##### Refers/Fixes

fixes #2433 
fixes #2460

### CODE REVIEW

File: packages/grant-explorer/src/features/common/PassportWidget.tsx
Comment (vacekj) [inline packages/grant-explorer/src/features/common/PassportWidget.tsx:66]: there's already a usePassport hook for this - we should reuse it.

Comment (bhargavaparoksham) [reply packages/grant-explorer/src/features/common/PassportWidget.tsx:66]: @vacekj  Yes, agree. I didn't reuse it initially because, the hook was just returning matching available or not state but not the actual score, which we need in the widget.

Will update the hook itself now as we've decided to remove the passport banner & updating the passport logic across all our apps.

**Comment by melissa-neira** [comment]
@bhargavaparoksham looks good to me but the fonts & font weights are a little bit different than in the original design.
The fonts we're using for the widget are: Modern Era & DM Mono. let me know if you need these fonts! you can also check for the font weights in figma [here](https://www.figma.com/file/1uxgU2jb0yyN11SUgxARmF/Explorer-S19?type=design&node-id=677-25056&mode=design&t=4a9nN8N0y7YiMtSp-4)

**Comment by michellema1208** [comment]
hey @bhargavaparoksham @gravityblast - the widget is looking AMAZING so far, great work!! 

Noticing that the calculations for donation impact aren't fully accurate -- see below for my wallet with score of 21.83 -- my donation impact is 50%, but it should be higher, as the range is score of 15-25, donation impact of 0-100% where 15=0% and 25=100%, scaled linearly. Let me know if you have questions! 

Screen Shot 2023-10-20 at 4 06 10 PM

**Comment by bhargavaparoksham** [comment]
@michellema1208 thanks for clarifying this. Currently In the ui I've implemented donation impact calc similar to colorCalc i.e 3 different slabs as below,
0-15 -> 0% impact
15-25 -> 50% impact
25+ -> 100% impact

I'll update this.

The question I have now, before i update it is, when the score is between 15 to 25 does donation impact, scale from O to 100 or does it scale from 50 to 100. Because in the ticket ACs it says it should scale from 50 to 100.

cc: @gravityblast

**Comment by michellema1208** [comment]
@bhargavaparoksham ah, great question! When the score is **between 15 to 25, the donation impact scales from 0-100%**. This was based off of a recent convo with @gravityblast, but wasn't updated in the ticket - thank you for catching that! 

Also, get some rest / enjoy your long weekend, Bhargav! I'll work with Andrea on Monday to get any work you have left over the finish line :)

File: packages/grant-explorer/src/features/api/passport.ts
Comment (vacekj) [inline packages/grant-explorer/src/features/api/passport.ts:48]: parse response with zod here to avoid wrong data and alert about api changes.

File: packages/grant-explorer/src/features/api/passport.ts
Comment (vacekj) [inline packages/grant-explorer/src/features/api/passport.ts:21]: let's use a string union here

**Comment by vacekj** [comment]
@gitcoinco/grantsstack-product fixed the issues you mentioned. a good address for testing is coday.eth for a 50 (!!!!!) passport score, and vitalik.eth for a meagre 20 passport score.

@melissa-neira I'm gonna need the Modern Era font, I see it's paid. I'll also fix the DM Mono font as part of this change.

What's left are the fonts as per ^.

**Review by michellema1208 — APPROVED** [review]
hey @vacekj - thanks so much for finishing this up! All of the changes look right to me. I have a small favor to ask to update the copy -- realizing that the current copy doesn't clarify the "Passport score of 15" threshold, and that's on me! 

New copy: "Your donation impact is calculated based on your Passport score. Scores higher than 15 will begin to be eligible for matching, and your donation impact scales as your Passport score increases. You can update your score by heading over to Passport."

**Comment by melissa-neira** [comment]
everything looks good to me so far! I'll approve once the fonts have been updated

File: packages/grant-explorer/src/features/api/passport.ts
Comment (bard) [inline packages/grant-explorer/src/features/api/passport.ts:74]: This could be done in the zod schema itself with `z.coerce.number()`

File: packages/grant-explorer/src/features/common/PassportWidget.tsx
Comment (bard) [inline packages/grant-explorer/src/features/common/PassportWidget.tsx:15]: `isXyz` would be a clearer name for boolean parameters than `xyz` (can't get mistaken for an action)

**Comment by melissa-neira** [comment]
@vacekj the padding and spacing of text is a little off on my screen
on staging:
Screenshot 2023-10-24 at 12 34 19
in figma:
Screenshot 2023-10-24 at 12 34 28

I'd try to increase the text width (the main copy) and make the widget a little wider and decrease the overall length

File: packages/common/src/index.ts
Comment (vacekj) [reply packages/common/src/index.ts:34]: I specifically picked nullish here, as optional doesn't cover the `null` case, which is present here when the score is loading.

File: packages/grant-explorer/src/features/api/passport.ts
Comment (vacekj) [reply packages/grant-explorer/src/features/api/passport.ts]: Is the plan to Make those Explicit via Boolean()? Because eslint complains when I do that, that it's supposedly unnecessary. Personally it's very clear what these are doing. I wish js had an idiomatic way to check for both null and undefined that wasn't `==`.

File: packages/grant-explorer/src/features/api/passport.ts
Comment (vacekj) [reply packages/grant-explorer/src/features/api/passport.ts]: Done.

File: packages/grant-explorer/src/features/common/PassportWidget.tsx
Comment (vacekj) [reply packages/grant-explorer/src/features/common/PassportWidget.tsx:15]: Done.

File: packages/grant-explorer/src/features/discovery/RoundCard.tsx
Comment (vacekj) [reply packages/grant-explorer/src/features/discovery/RoundCard.tsx:111]: Yeah, the undefined propagates down and displays some default or empty values. This is a bugfix for a crash. The proper solution for another ticket is to investigate the upstream types of this to make sure that parsing is in place on the domain boundary and the types reflect that.

File: packages/grant-explorer/src/features/round/ViewCartPage/RoundInCart.tsx
Comment (vacekj) [reply packages/grant-explorer/src/features/round/ViewCartPage/RoundInCart.tsx:14]: Renamed and also made typesafe.

File: packages/grant-explorer/src/features/round/ViewCartPage/RoundInCart.tsx
Comment (vacekj) [reply packages/grant-explorer/src/features/round/ViewCartPage/RoundInCart.tsx:64]: Fixed.

File: packages/grant-explorer/src/features/api/passport.ts
Comment (bard) [reply packages/grant-explorer/src/features/api/passport.ts]: I usually enable https://palantir.github.io/tslint/rules/strict-boolean-expressions/. 

It's primarily a matter of guarding against conditions greedier than expected. Maybe you've found yourself trying to figure why something like this would execute, when `s` is `string | undefined` and you've already verified that it's not `undefined`:

```ts
if (!s) {
  console.log('whooops')
}
```

Then you're at your wits' ends and finally decide to `console.log(s)`. Turns out that `s` was an empty string, which is falsy. Whooops.

(Same thing when `s` is a `number` and `0` starts triggering all your `null | undefined` conditions.)

I get the pain of writing `if (a === null | a === undefined)`. For cases like these I like to use `isPresent` from [ts-is-present](https://www.npmjs.com/package/ts-is-present).

**Comment by vacekj** [comment]
@melissa-neira 
The widget dimensions are exactly the same as in figma. the only thing that's different is the copy, and the text size. in figma, the text size is 10px. In code, it was previously 14px, now 12px. I would like to push back against this small of a font size and go with at least 12px (current) or bigger. 10px is barely readable for me on a 27" 4K monitor, and I have close to 20/20 vision, can't imagine someone with a vision deficiency reading this easily.

What I've done is increased the line height to align with figma + font size increase (18px figma, 20px code) and checked all the margin and padding to align with figma.

Please do note that although Figma runs in a browser, they utilize a completely separate rendering engine, so there might be small variations between the actual browser and figma, esp. in text rendering and wrapping.

Lmk your thought and if I should decrease the font size to 10px or leave it as-is.

**Comment by melissa-neira** [comment]
@vacekj good call out! yes, let's go ahead with that then :) thank you Josef!

**Review by melissa-neira — APPROVED** [review]

**Comment by bard** [comment]
@bhargavaparoksham this was probably opened before the PR checklist got merged. I added it to the PR description.

**Review by vacekj — APPROVED** [review]

**Pull request merged**
